app/frontend/src/wells/store/index.js

```diff
@@ -225,42 +222,21 @@ const wellsStore = {
         }
         commit(SET_SEARCH_RESULTS, null)
         commit(SET_SEARCH_RESULT_COUNT, 0)
-      }).finally(() => {
-        commit(SET_PENDING_SEARCH, null)
       })
-    },
-    [SEARCH_WELL_LOCATIONS] ({ commit, state }, { bounded = false }) {
-      if (state.locationPendingSearch) {
-        state.locationPendingSearch.cancel()
-      }
-      commit(SET_LOCATION_PENDING_SEARCH, axios.CancelToken.source())
-
-      const params = { ...state.searchParams }
-
-      if (Object.entries(state.searchResultFilters).length > 0) {
-        params['filter_group'] = JSON.stringify(state.searchResultFilters)
-      }
-
-      // if triggering the search using the map, the search will be restricted to
-      // the visible map bounds
-      if (bounded) {
-        Object.assign(params, state.searchBounds)
-      }
 
-      ApiService.query('wells/locations', params, { cancelToken: state.locationPendingSearch.token }).then((response) => {
-        const locations = response.data.map((well) => {
-          return [well.latitude, well.longitude, well.well_tag_number, well.identification_plate_number]
-        })
+      const locationApiQuery = ApiService.query('wells/locations', locationParams, { cancelToken: cancelSource.token }).then((response) => {
         commit(SET_LOCATION_ERRORS, {})
-        commit(SET_LOCATION_SEARCH_RESULTS, locations)
-        commit(SET_LOCATION_SEARCH_RESULT_COUNT, response.data.count)
+        commit(SET_LOCATION_SEARCH_RESULTS, response.data)
       }).catch((err) => {
         if (err.response && err.response.data) {
           commit(SET_LOCATION_ERRORS, err.response.data)
         }
         commit(SET_LOCATION_SEARCH_RESULTS, [])
-      }).finally(() => {
-        commit(SET_LOCATION_PENDING_SEARCH, null)
+      })
+
+      // Clear pending after everything completes.
+      Promise.all([wellApiQuery, locationApiQuery]).then(() => {
+        commit(SET_PENDING_SEARCH, null)
```

Comment:
if one of the requests results in an error, will this get stuck in a pending state?

Reply:
I've revised to use `finally`, but I think this was already working. Because each Axios query has a `catch`, error responses still result in successful promises here. It is confusing though! Comment added in d7f515a.